inc/features/blocks/personalization-variant/edit.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+const { InnerBlocks } = wp.blockEditor;
+const { compose } = wp.compose;
+const { withSelect, withDispatch } = wp.data;
+
+const Edit = ( {
+	hasChildBlocks,
+	isSelected,
+	selectParent,
+} ) => {
+	// Autofocus experience block variant parent on select.
+	if ( isSelected ) {
+		selectParent();
+	}
+
+	return (
+		<InnerBlocks
+			renderAppender={
+				hasChildBlocks
+					? undefined
+					: () => <InnerBlocks.ButtonBlockAppender />
```

Comment:
This is really confusing; should be moved out of this statement and reworked to be a bit more understandable. `undefined` also seems like a very strange value to be passing here.

Something like this would be a bit more understandable:

```js
const props = {};
if ( ! hasChildBlocks ) {
    // If we don't have any child blocks, append button block (or something)
    props.renderAppender = () => <InnerBlocks.ButtonBlockAppender />;
}

<InnerBlocks { ...props } />
```

Reply:
Yeah copied from the core columns block, makes sense to simplify it like this.